packages/@aws-cdk/aws-apprunner/lib/vpc-connector.ts

```diff
@@ -0,0 +1,135 @@
+
+  /**
+    * A list of subnets that App Runner should use when it associates the service with a custom Amazon VPC.
+    */
+  readonly subnets: ec2.ISubnet[];
+
+  /**
+    * The name for the VpcConnector.
+    *
+    * @default - a name generated by CloudFormation
+    */
+  readonly vpcConnectorName?: string;
+}
+
+/**
+ * Attributes for the App Runner VPC Connector
+ */
+export interface VpcConnectorAttributes {
+  /**
+   * The name of the VPC connector.
+   */
+  readonly vpcConnectorName: string;
+
+  /**
+   * The ARN of the VPC connector.
+   */
+  readonly vpcConnectorArn: string;
+
+  /**
+   * The revision of the VPC connector.
+   */
+  readonly vpcConnectorRevision: number;
+}
+
+/**
+ * Represents the App Runner VPC Connector.
+ */
+export interface IVpcConnector extends cdk.IResource {
+  /**
+   * The Name of the VPC connector.
+   */
+  readonly vpcConnectorName: string;
+
+  /**
+   * The ARN of the VPC connector.
+   */
+  readonly vpcConnectorArn: string;
+}
+
+/**
+ * The App Runner VPC Connector
+ */
+export class VpcConnector extends cdk.Resource {
+  /**
+   * Import from VPC connector name.
+   */
+  public static fromVpcConnectorName(scope: Construct, id: string, vpcConnectorName: string): IVpcConnector {
+    class Import extends cdk.Resource {
+      public vpcConnectorName = vpcConnectorName;
+      public vpcConnectorArn = cdk.Stack.of(this).formatArn({
+        resource: 'vpcconnector',
+        service: 'apprunner',
+        resourceName: vpcConnectorName,
+      })
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * Import from VPC connector attributes.
+   */
+  public static fromServiceAttributes(scope: Construct, id: string, attrs: VpcConnectorAttributes): IVpcConnector {
+    const vpcConnectorArn = attrs.vpcConnectorArn;
+    const vpcConnectorName = attrs.vpcConnectorName;
+    const vpcConnectorRevision = attrs.vpcConnectorRevision;
+
+    class Import extends cdk.Resource {
+      public readonly vpcConnectorArn = vpcConnectorArn
+      public readonly vpcConnectorName = vpcConnectorName
+      public readonly vpcConnectorRevision = vpcConnectorRevision
+    }
+
+    return new Import(scope, id);
+  }
+  private readonly props: VpcConnectorProps;
+
+  /**
+    * The ARN of the VPC connector.
+    * @attribute
+    */
+  readonly vpcConnectorArn: string;
+
+  /**
+   * The revision of the VPC connector.
+   * @attribute
+   */
+  readonly vpcConnectorRevision: number;
+
+  /**
+    * The name of the VPC connector.
+    * @attribute
+    */
+  readonly vpcConnectorName: string;
+
+  public constructor(scope: Construct, id: string, props: VpcConnectorProps) {
+    super(scope, id);
+
+    this.props = props;
+
+    const resource = new CfnVpcConnector(this, 'VpcConnector', {
+      subnets: this.props.subnets.map(subnet => subnet.subnetId),
+      securityGroups: this.props.securityGroups?.map(securityGroup => securityGroup.securityGroupId),
+      vpcConnectorName: this.props.vpcConnectorName,
```

Comment:
instead of assigning the name here, can you use physical name?

```ts
super(scope, id, {
  physicalName: props.vpcConnectorName,
});

vpcConnectorName: this.physicalName,
```